demo_notebook/openspoor_example.ipynb

```diff
@@ -32,7 +31,46 @@
     "import os\n",
     "import openspoor\n",
     "from openspoor import SpoortakMapservices, PUICMapservices\n",
-    "from openspoor import TransformerCoordinatesToSpoor, TransformerGeocodeToCoordinates, TransformerSpoortakToCoordinates"
+    "from openspoor import TransformerCoordinatesToSpoor, TransformerGeocodeToCoordinates, TransformerSpoortakToCoordinates\n",
+    "from openspoor.visualisations.spoorkaart import SpoorKaart, PlottingDataFrame"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Demo 1 - Set up spoorkaart and add points to plot\n",
+    "\n",
+    "Making a map consists of setting up a SpoorKaart object and adding the objects you wish to plot to it. These objects can be given in Pandas DataFrames, which are displayed on a map with added aerial photographs of the Dutch tracks and zoomed to the location of interest.\n",
+    "Optionally, these outputs can be saved as .html files which can then be shared or used in applications.\n",
+    "Within notebooks, Spoorkaart objects are displayed if their value is requested at the end of a cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with SpoorKaart() as m:\n",
+    "    # Add pretty markers with colors and custom Font Awesome icons at custom locations\n",
+    "    PlottingDataFrame({'lat': [52.08, 52.093],\n",
+    "                       'lon': [5.119, 5.107],\n",
+    "                       'value': [1, 2],\n",
+    "                       \"marker\": ['train', \"eye\"]},\n",
+    "                      colors={'value': {(0, 1.5): 'blue', (1.5, 3.0): 'orange'}},\n",
+    "                      marker_column=\"marker\").add_to(m)\n",
+    "    # Add clickable markers with arrows to indicate directions at locations\n",
+    "    PlottingDataFrame({'lat': [52.0874117], 'lon': [5.1156098], \"rotation\": [245], 'location': ['ProRail Entrance']},\n",
+    "                      colors='purple',\n",
+    "                      popup=['location'],\n",
+    "                      rotation_column=\"rotation\").add_to(m)\n",
+    "    # Plot clickable circle(s) on a map\n",
+    "    PlottingDataFrame({'lat': [52.086420], 'lon': [5.113101], 'radius': [10], 'object': ['switch']},\n",
+    "                      popup=['object'],\n",
+    "                      markertype='circle',\n",
+    "                      radius_column=\"radius\").add_to(m)\n",
+    "m"
```

Comment:
needs an extra tab (as to be with in the scope of the with statement.
Python is allowed to clean up any resources assosiated with m as soon as you exit the with

Reply:
This does work; as the __exit__ statement returns the value of m itself m keeps on existing, even after the with statement. I do agree that this is not clear based on the code and I think this logic can be refactored into something that does not use the with statement anymore. This would require quite a large syntax change however, so I think it would be better to pick this up in a new PR.